### ISSUE TITLE
[Git master] Still running after 25 minutes?

### ISSUE THREAD

**Issue by hartwork** [NASA-SW-VnV/ikos#149]
I'm not sure if this is a bug: ikos kept running for 25 minutes on this code before I killed it. Is this to be expected? I played with some of the widening  and narrowing parameters but was not able to make ikos do fewer terations, either. To reproduce:

```console
$ wget https://github.com/libexpat/libexpat/releases/download/R_2_2_7/expat-2.2.7.tar.xz
$ tar xf expat-2.2.7.tar.xz 
$ cd expat-2.2.7/
$ time PYTHONPATH=/tmp/tmp.6VxVXsOPnE/ikos/install/lib64/python3.6/site-packages/ /tmp/tmp.6VxVXsOPnE/ikos/install/bin/ikos -Ilib -e XML_Parse -D HAVE_GETRANDOM lib/xmlparse.c
[*] Compiling lib/xmlparse.c
[*] Running ikos preprocessor
[*] Running ikos analyzer
[*] Translating LLVM bitcode to AR
[*] Running liveness analysis
[*] Running widening hint analysis
[*] Running interprocedural value analysis
[*] Analyzing entry point 'XML_Parse'
> lib/xmlparse.c:1721:45: Analyzing called function 'startParsing'
> lib/xmlparse.c:859:12: Analyzing called function 'setContext'
> lib/xmlparse.c:5969:10: Analyzing loop [iteration 2↑]
> lib/xmlparse.c:5990:25: Analyzing called function 'lookup'
> lib/xmlparse.c:6391:23: Analyzing called function 'hash'
> lib/xmlparse.c:6368:3: Analyzing called function 'sip24_update'
> lib/siphash.h:0:0: Analyzing loop [iteration 1↑]
> lib/siphash.h:217:5: Analyzing loop [iteration 10667701↑]^C
real    24m59.103s
user    0m0.463s
sys     0m0.048s
```

This is ikos commit deaa9b47ae69b9af8b622d75cf0a04c66e2f7403.

**Comment by arthaud** [comment]
It looks like a bug, looking at the number of iterations performed. Let me investigate.

**Comment by arthaud** [comment]
This is fixed in f903566b2bf1d8d05384d0dbb47f63b509e934a6.

By the way, you will get a lot of false positives by analyzing a single file (`xmlparse.c`), because the analyzer will be missing the implementation of most functions. Consider using `ikos-scan` and `ikos-scan-extract` to generate the llvm bitcode of the whole library (see [README.md](https://github.com/NASA-SW-VnV/ikos/blob/master/analyzer/README.md#analyze-a-whole-project-with-ikos-scan))

Since you are analyzing a library, there will be no `main` function. A workaround is to create a fake `main` that calls the library. See for instance: https://github.com/ANSSI-FR/x509-parser/blob/9132d7dea8efce1b59549110b25608daab622aca/src/x509-parser.c#L9253
